l10n_it_ricevute_bancarie/wizard/wizard_riba_file_export.py

```diff
@@ -234,16 +235,17 @@ def act_getfile(self):
             debitor_address = line.partner_id
             debitor_street = debitor_address.street or ''
             debitor_zip = debitor_address.zip or ''
+            debit_iban = debit_bank.acc_number.replace(" ", "") \
+                if debit_bank.acc_number else None
             if debit_bank.bank_abi and debit_bank.bank_cab:
                 debit_abi = debit_bank.bank_abi
                 debit_cab = debit_bank.bank_cab
-            elif debit_bank.acc_number:
-                debit_iban = debit_bank.acc_number.replace(" ", "")
+            elif debit_iban and len(debit_iban) == 27:
```

Comment:
È possibile che l'utente inserisca un IBAN fasullo solo ai fini dell'emissione della riba (l'ho visto fare) e dato che non è considerata la parte relativa al conto corrente e non è obbligatorio che l'IBAN sia completo, è una limitazione aggiuntiva senza utilità.

Reply:
Ma in quel caso non dovrebbe a questo punto usare abi/cab e non sporcare iban ... oltretutto l'iban viene controllato e validato da base_iban da cui questo modulo dipende. Se non metti 27 cosa metti? se metto iban di 7 caratteri ottengo un errore ... in presenza di iban farlocchi il risultato è aleatorio. Bhu ...

Reply:
Concordo che il campo IBAN se inserito è bene che sia un IBAN

Reply:
[Questo](https://github.com/odoo/odoo/blob/babada244b17a0dda479508390bd221b7c17474a/addons/base_iban/models/res_partner_bank.py#L58) è il punto in cui nel codice viene verificato se è un IBAN valido e nel caso impostato il tipo ad IBAN, altrimenti non succede alcunchè, rimane un conto normale.
Se si vuole modificare il comportamento del sistema, secondo me va fatto a monte, non nel momento di emissione di una riba mettendo di fronte l'utente ad un problema nel momento in cui sta completando l'operazione.

Reply:
Concordo di mettere un controllo che il conto inserito sia almeno di 15 caratteri